### PR TITLE
Don't run RE2 and GMP builds at the same time

### PR DESCRIPTION
**Pull request by mppf**
I noticed that in some nightly testing configurations, the RE2 and GMP builds are running at the same time. That leads to confusing output. This PR adjusts the top-level makefile to indicate that its targets shouldn't run in parallel. The sub-targets still run in parallel.

Reviewed by @ronawho - thanks!

### CODE REVIEW

**Comment by ronawho** [comment]
Parallel make output is going to be confusing no matter what, so it seems like a shame to hurt build times just for slightly more readable output.  Can you gather timings on chapcs or something for how much this impacts build times? (my concern is how much this will impact module build time)

**Comment by mppf** [comment]
|                                                | master  |  this PR |
| ---------------------------- | -------- | ------- |
| make -j4 third-party-try-opt | 77s        |  84s  |
| make -j4                                | 209s      | 219s |

I don't think 10s out of 200s is "worth it" for having mixed output during configure of these packages.
I think the speculative build logic is already weird and hard to follow enough that it's worth the simplification.
